native/tools/swig-config/tutorial/index.md

```diff
@@ -689,6 +711,95 @@ Build and run project
+
+In `native/tools/swig-config/cocos.i`, there are:
+
+```c++
+%attribute_writeonly(cc::ICanvasRenderingContext2D, float, width, setWidth);
+%attribute_writeonly(cc::ICanvasRenderingContext2D, float, height, setHeight);
+%attribute_writeonly(cc::ICanvasRenderingContext2D, float, lineWidth, setLineWidth);
+%attribute_writeonly(cc::ICanvasRenderingContext2D, ccstd::string&, fillStyle, setFillStyle);
+%attribute_writeonly(cc::ICanvasRenderingContext2D, ccstd::string&, font, setFont);
+```
+
+This is the similar functionality like which in JS:
+
+```javascript
+Object.defineProperty(MyNewClass.prototype, 'width', {
+  configurable: true,
+  enumerable: true,
+  set(v) {
+    this._width = v;
+  },
+	// No get() for property
+});
+```
+
+#### Reference type
+
+If C++ `get` and `set` function return reference type, don't forget to add `&` suffix in %attribute or %attribute_writeonly directives. The following `ccstd::string&` is an example.
+
+```c++
+%attribute_writeonly(cc::ICanvasRenderingContext2D, ccstd::string&, fillStyle, setFillStyle);
+```
+
+If `&` is missing, temporary `ccstd::string` objects will be created while binding functions are invoked.
+
+#### %arg() directive
+
+Sometimes, the type of C++ variable is a describled by C++ template, for instance:
+
+```c++
+class MyNewClass {
+  public:
+		const std::map<std::string, std::string>& getConfig() const { return _config; }
+  	void setConfig(const std::map<std::string, std::string> &config) { _config = config; }
+  private:
+  	std::map<std::string, std::string> _config;
+};
+```
+
+We may write an `%attribute` in `.i` file like:
+
+```c++
+%attribute(MyNewClass, std::map<std::string, std::string>&, config, getConfig, setConfig);
+```
+
+You will get an error while invoking `node genbindings.js`.
+
+```
+Error: Macro '%attribute_custom' expects 7 arguments
+```
+
+This is because `swig` doesn't know how to deal with comma (`,`) in `std::map<std::string, std::string>&`, it will split it to two parts:
+
+1. std::map<std::string
+2. std::string>&
+
+Therefore, this line of %attribute directive will be parsed with 6 arguments instead of 5.
+
+To avoid making `swig` confused, we need to use `%arg` directive to tell `swig` that `std::map<std::string, std::string>&` is a whole thing.
+
+```c++
+%attribute(MyNewClass, %arg(std::map<std::string, std::string>&), config, getConfig, setConfig);
+```
+
+Re-run `node genbindings.js`, the error should have gone.
+
+#### Don't add `const` 
+
+In the previous sample, `%arg(std::map<std::string, std::string>&)` is used as C++ type in %attribue directive. You may consider to add a const prefix before `std::map` like `%arg(const std::map<std::string, std::string>&)`. If you do that, you will make a readyonly `config` property which only bind `MyNewClass::getConfig`. That's obviously not what we expect. If we need a readonly property, just don't assign a `set` function. 
```

Comment:
In the previous sample -> In above sample

Reply:
is used as C++ type -> is used as a C++ data type

Reply:
Any differences about `previous` and `above`?  I think they're both ok here.

Reply:
And while we uses `above`, should we drop `the` before it?
Is it `In the above sample` correct or `In above sample`?
I think  `the`  word should not be dropped since it describe the `exact` sample in the previous section.